module/support/optimisation/NSGA2Evaluator/src/tasks/WalkEvaluator.cpp

```diff
@@ -0,0 +1,206 @@
+
+                double trialDuration = simTime - trialStartTime;
+                NUClear::log<NUClear::DEBUG>("Trial ran for", trialDuration);
+                NUClear::log<NUClear::DEBUG>("SendFitnessScores for generation", generation, "individual", individual);
+                NUClear::log<NUClear::DEBUG>("    scores:", scores[0], scores[1]);
+                NUClear::log<NUClear::DEBUG>("    constraints:", constraints[0], constraints[1]);
+
+                // Create the fitness scores message based on the given results and emit it back to the Optimiser
+                std::unique_ptr<NSGA2FitnessScores> fitnessScores = std::make_unique<NSGA2FitnessScores>();
+                fitnessScores->id                                 = individual;
+                fitnessScores->generation                         = generation;
+                fitnessScores->objScore                           = scores;
+                fitnessScores->constraints                        = constraints;
+                return fitnessScores;
+            }
+            
+            std::vector<double> WalkEvaluator::calculateScores() {
+                auto robotDistanceTravelled = std::fabs(initialRobotPosition.x() - robotPosition.x());
+                return {
+                    maxFieldPlaneSway,            // For now, we want to reduce this
+                    1.0 / robotDistanceTravelled  // 1/x since the NSGA2 optimiser is a minimiser
+                };
+            }
+
+            std::vector<double> WalkEvaluator::calculateConstraints(double simTime) {
+                // Convert trial duration limit to ms, add 1 for overhead
+                const auto overhead = std::chrono::seconds(1);
+                double max_trial_duration =
+                    (std::chrono::duration_cast<std::chrono::milliseconds>(trial_duration_limit + overhead)).count();
+                double trialDuration = simTime - trialStartTime;
+                return {
+                    trialDuration - max_trial_duration,  // Punish for falling over, based on how long the trial took
+                                                         // (more negative is worse)
+                    0.0                                  // Second constraint unused, fixed to 0
+                };
+            }
+
+            std::vector<double> WalkEvaluator::constraintsNotViolated() {
+                return {
+                    0,  // Robot didn't fall
+                    0   // Second constraint unused, fixed to 0
+                };
+            }
+
+
+            bool WalkEvaluator::checkForFall(const RawSensors& sensors) {
+                bool fallen        = false;
+                auto accelerometer = sensors.accelerometer;
+
+                if ((std::fabs(accelerometer.x()) > 9.2 || std::fabs(accelerometer.y()) > 9.2)
+                    && std::fabs(accelerometer.z()) < 0.5) {
```

Comment:
Where are these values coming from? They look like they should be in a config file?

Reply:
There are no config files. These are raw sensor reading checking is the robot has fell over

Reply:
But where are `9.2` and `0.5` coming from? They seem like they should be in the module's configuration file. Shouldn't hardcode values